client/data/settings/hooks.js

```diff
@@ -4,6 +4,56 @@ import { STORE_NAME } from '../constants';
 
 const EMPTY_ARR = [];
 
+const makeSettingsHookFromUpdateHandler = (
+	fieldName,
+	updateHandler,
+	fieldDefaultValue = false
+) => () => {
+	const { updateSettingsValues } = useDispatch( STORE_NAME );
+
+	const field = useSelect(
+		( select ) => {
+			const { getSettings } = select( STORE_NAME );
+
+			return getSettings()[ fieldName ] || fieldDefaultValue;
+		},
+		[ fieldName, fieldDefaultValue ]
+	);
```

Comment:
LMK if I'm missing something, but could this be replaced by `makeReadOnlySettingsHook`?
```
const value = makeReadOnlySettingsHook( fieldName, fieldDefaultValue );
```